src/connections/allContacts.js

```diff
@@ -35,6 +39,37 @@ export const createContact = (client, attributes) =>
 
 export const updateContact = (client, contact) => client.save(contact)
 
+const isContactSynced = contact => {
+  return Object.keys(contact.cozyMetadata?.sync || {}).length > 0
+}
+
+const getGroupRelationshipLength = contact => {
+  return contact.relationships?.groups?.data.length
+}
+
+/**
+ * Delete trashed contacts without linked to an account
+ *
+ * @param {Object} client - CozyClient
+ */
+export const deleteTrashedContacts = async client => {
+  const contactContactsTrashedQuery = buildContactsTrashedQuery()
+
+  const { data: trashedContacts } = await client.query(
+    contactContactsTrashedQuery.definition(),
+    contactContactsTrashedQuery.options
+  )
+
+  const trashedContactsNotLinkedToAccounts = trashedContacts.filter(
+    contact => !isContactSynced(contact)
+  )
+
+  if (trashedContactsNotLinkedToAccounts.length > 0) {
+    const contactCollection = client.collection(DOCTYPE_CONTACTS)
+    await contactCollection.destroyAll(trashedContactsNotLinkedToAccounts)
```

Comment:
Can you create an issue on cozy-client to make the destroyAll() accessible from the `client.` In order to not call the `collection` ? 🙏